yarn-project/cli/src/unbox.ts

```diff
@@ -0,0 +1,320 @@
+ * @returns The path to the downloaded contract.
+ */
+async function downloadContractAndBoxFromGithub(
+  contractName: string = 'PrivateToken',
+  outputPath: string,
+  log: LogFn,
+): Promise<void> {
+  // small string conversion, in the ABI the contract name looks like PrivateToken
+  // but in the repostory it looks like private_token
+  const snakeCaseContractName = contractNameToFolder(contractName);
+  // source noir files for the contract are in this folder
+  const contractFolder = `${NOIR_CONTRACTS_PATH}/${snakeCaseContractName}_contract`;
+  await downloadNoirFilesFromGithub(contractFolder, snakeCaseContractName, outputPath, log);
+
+  log(`Downloaded '@aztex/boxes/${snakeCaseContractName}' to ${outputPath}`);
+  return;
+}
+
+/**
+ *
+ * @param data - in memory unzipped clone of a github repo
+ * @param repositoryFolderPath - folder to copy from github repo
+ * @param localOutputPath - local path to copy to
+ */
+async function copyFolderFromGithub(data: JSZip, repositoryFolderPath: string, localOutputPath: string, log: LogFn) {
+  log('downloading from github:', repositoryFolderPath);
+  const repositoryDirectories = Object.values(data.files).filter(file => {
+    return file.dir && file.name.startsWith(repositoryFolderPath);
+  });
+
+  for (const directory of repositoryDirectories) {
+    const relativePath = directory.name.replace(repositoryFolderPath, '');
+    const targetPath = `${localOutputPath}/${relativePath}`;
+    await fs.mkdir(targetPath, { recursive: true });
+  }
+
+  const starterFiles = Object.values(data.files).filter(file => {
+    return !file.dir && file.name.startsWith(repositoryFolderPath);
+  });
+
+  for (const file of starterFiles) {
+    const relativePath = file.name.replace(repositoryFolderPath, '');
+    const targetPath = `${localOutputPath}/${relativePath}`;
+    const content = await file.async('nodebuffer');
+    await fs.writeFile(targetPath, content);
+  }
+}
+
+/**
+ * Not flexible at at all, but quick fix to download a noir smart contract from our
+ * monorepo on github.  this will copy over the `yarn-projects/boxes/{contract_name}` folder
+ * as well as the specified `directoryPath` if the box doesn't include source code
+ * `directoryPath` should point to a single noir contract in `yarn-projects/noir-contracts/src/contracts/...`
+ * @param directoryPath - path to a noir contract's source code (folder) in the github repo
+ * @param outputPath - local path that we will copy the noir contracts and web3 starter kit to
+ * @returns
+ */
+async function downloadNoirFilesFromGithub(
+  directoryPath: string,
+  snakeCaseContractName: string,
+  outputPath: string,
+  log: LogFn,
+): Promise<void> {
+  const owner = GITHUB_OWNER;
+  const repo = GITHUB_REPO;
+  // Step 1: Fetch the monorepo ZIP from GitHub, hardcoded to the master branch
+  const url = `https://github.com/${owner}/${repo}/archive/refs/heads/master.zip`;
```

Comment:
Maybe download from the same version as cli. Same for the dependencies in Nargo.toml and the sandbox version in package.json. That's probably the only 2 files we need to generate dynamically when this command is ran. This will prevent the issue we encountered the other day, where the simulator apis in the sandbox are not compatible with the oracle calls from the noir contract.

Reply:
good call!  updated

Reply:
I believe noir is actively working on a tag that would indicate compatibility with the aztec-packages semver, but we can't automatically match versions at the moment